### ISSUE TITLE
Large Ring not detected during OBMolRingIter

### ISSUE THREAD

**Issue by nate-russell** [openbabel/openbabel#1858]
I believe this to be a bug with Open Babel

## Environment Information
Open Babel version:  openbabel  2.4.1  py36_5    openbabel (from conda list)
Operating system and version: Ubuntu 14.04.5 LTS (GNU/Linux 3.19.0-58-generic x86_64)

## Expected Behavior
molecule with SMILES: "OC[C@@H]1OCC[C@H](C)CCC[C@H](C)CCC[C@@H](C)CCC[C@H](C)CC[C@@H](C)CCC[C@H](C)CCC[C@@H](C)CCC[C@@H](C)CCOC[C@H](OCC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CCOC1)C)C)C)C)C)C)C)C)CO[C@@H]1O[C@H](CO[C@@H]2O[C@H](CO)[C@H]([C@@H]([C@H]2O)O)O)[C@H]([C@@H]([C@H]1O)O)O"

Has 2 six membered rings and one very large ring (macrocycle).  
## Actual Behavior
When iterating over ring instances with OBMolRingIter it only returns the two six membered rings. I think it should also return the macrocycle

## Steps to Reproduce
```python
import openbabel as ob
import pybel

smi = "OC[C@@H]1OCC[C@H](C)CCC[C@H](C)CCC[C@@H](C)CCC[C@H](C)CC[C@@H](C)CCC[C@H](C)CCC[C@@H](C)CCC[C@@H](C)CCOC[C@H](OCC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CCOC1)C)C)C)C)C)C)C)C)CO[C@@H]1O[C@H](CO[C@@H]2O[C@H](CO)[C@H]([C@@H]([C@H]2O)O)O)[C@H]([C@@H]([C@H]1O)O)O"

mol = pybel.readstring("CAN", smi)

for obring in ob.OBMolRingIter(mol.OBMol):
    print(obring.Size())
```

**Comment by ghutchis** [comment]
I don't believe this to be a bug. Open Babel uses the SSSR idiom - Smallest Set of Smallest Rings.

It does not include an enumeration of all possible rings, since that would be an exponential-time algorithm.

**Comment by nate-russell** [comment]
Then I think there should be some documentation that tells users the max ring size looked for in SSSR or even better allows them to choose because a molecule like 
```
 smi = "OC[C@@H]1OCCCCCC(OCC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CCOC1)C)C)C)C)C)C)C)C)CO[C@@H]1O[C@H](CO[C@@H]2O[C@H](CO)[C@H]([C@@H]([C@H]2O)O)O)[C@H]([C@@H]([C@H]1O)O)O"
``` 
which has rings of size 6,6 and 43, all 3 rings are reported in SSSR. But add one more carbon to the macro cycle like 
```
smi = "OC[C@@H]1OCCCCCCC(OCC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CC[C@@H](CCC[C@@H](CCC[C@@H](CCC[C@@H](CCOC1)C)C)C)C)C)C)C)C)CO[C@@H]1O[C@H](CO[C@@H]2O[C@H](CO)[C@H]([C@@H]([C@H]2O)O)O)[C@H]([C@@H]([C@H]1O)O)O"
```
 and it only returns the 2 six member-ed rings.

**Comment by baoilleach** [comment]
I agree - it's a documentation bug, and I'm marking it as such. OBMolRingIter says "Iterate over all rings in an OBMol.". That's not what it does. It iterates over the SSSR.

If you are instead interested in whether an atom is in a particular sized ring, you should take a different approach.

**Comment by jeffjanes** [comment]
On Wed, Jul 4, 2018 at 4:34 AM, baoilleach <notifications@github.com> wrote:

> I agree - it's a documentation bug, and I'm marking it as such.
> OBMolRingIter says "Iterate over all rings in an OBMol.". That's not what
> it does. It iterates over the SSSR.
>

Does it really iterate over the SSSR correctly?  I would think the SSSR for
this molecule still contains 3 items in it.  Returning 2 items is wrong.

Cheers,

Jeff

**Comment by baoilleach** [comment]
Apologies - yes, you're right. I hadn't previously viewed it. We clearly bail out at a certain ring size.
